relay/chunkstore/chunk_writer.go

```diff
@@ -85,3 +104,26 @@ func (c *chunkWriter) PutChunkCoefficients(
 		FragmentSizeBytes:   uint32(c.fragmentSize),
 	}, nil
 }
+
+func (c *chunkWriter) proofExists(ctx context.Context, blobKey corev2.BlobKey) bool {
+	size, err := c.s3Client.HeadObject(ctx, c.bucketName, s3.ScopedProofKey(blobKey))
+	if err == nil && size != nil && *size > 0 {
+		return true
+	}
+
+	return false
+}
+
+func (c *chunkWriter) coefficientsExists(ctx context.Context, blobKey corev2.BlobKey, dataSize int64) bool {
+	objs, err := c.s3Client.ListObjects(ctx, c.bucketName, s3.ScopedChunkKey(blobKey))
```

Comment:
What do you think about adding a log for the latency here so that we can double check the performance?

Reply:
I was planning on adding metrics for all v2 components as a follow up (none of them export any metrics rn). I'll add this as part of those. Does that work?

Reply:
Yep!